models/errors.go

```diff
@@ -32,18 +33,31 @@ type VersionConflictError struct {
 
 // BadParameterError means that a parameter was not as required
 type BadParameterError struct {
-	parameter string
-	value     interface{}
+	parameter        string
+	value            interface{}
+	expectedValue    interface{}
+	hasExpectedValue bool
```

Comment:
Well, it could be that `nil` is actual an expected value, hence the additional value.